### ISSUE TITLE
Installing forte - cmake configuration from psi4

### ISSUE THREAD

**Issue by jfraxanet** [evangelistalab/forte#301]
Hello, 

I am encountering some problems when trying to install forte in Linux.
These are the steps that I am following:

- First of all, I made sure I have cmake 3.20 and Pybind11

- Then I install psi4 in a new enviroment using conda as follows
`conda create -n p4env psi4 -c psi4`
and I activate the environment
`conda activate p4env`

- I clone the forte github repository and open the setup.py file, where I change the line
`self.ambitpath = None`
to
`self.ambitpath =  "home/joana/anaconda3/envs/p4dev"`
where my ambit folder is. 

When I run ` python setup.py build_ext --debug develop` I get the following error:

> running build_ext

>  Forte compilation options

    BUILD_TYPE = Debug
    AMBITPATH = home/joana/anaconda3/envs/p4dev
    MAX_DET_ORB = 64
    ENABLE_CODECOV = OFF
    CMAKE_CONFIG_OPTIONS = 
    CMAKE_BUILD_OPTIONS = 
    NPROCS = 16

> CMake Error: Unknown argument -c
  CMake Error: Run 'cmake --help' for all supported options.
  Traceback (most recent call last):
  File "/home/joana/forte/setup.py", line 154, in <module>
    setup(
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 148, in setup
    return run_commands(dist)
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/site-packages/setuptools/_distutils/core.py", line 163, in run_commands
    dist.run_commands()
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 967, in run_commands
    self.run_command(cmd)
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/site-packages/setuptools/dist.py", line 1214, in run_command
    super().run_command(command)
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/site-packages/setuptools/_distutils/dist.py", line 986, in run_command
    cmd_obj.run()
  File "/home/joana/forte/setup.py", line 58, in run
    self.build_extension(ext)
  File "/home/joana/forte/setup.py", line 114, in build_extension
    subprocess.check_call(["cmake"] + cmake_args)
  File "/home/joana/anaconda3/envs/p4env/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '"psi4-dev"', 'via', '`conda', 'install', 'psi4-dev', '-c', 'psi4[/label/dev]`,', 'then', 'reissue', 'command.', '-Dambit_DIR=home/joana/anaconda3/envs/p4dev/share/cmake/ambit', '-DCMAKE_BUILD_TYPE=Debug', '-DMAX_DET_ORB=64', '-DENABLE_CODECOV=OFF', '-DENABLE_ForteTests=TRUE']' returned non-zero exit status 1.

It seems that the cmake configuration, which is grabed from psi4, is incorrect. I tryied installing psi4 differently, but cmake still fails and I guess it should work with the conda installation. Is there some way around this? 

Thank you!

Joana

**Comment by jfraxanet** [comment]
I have also tried to just get rid of the command '-c', but then I still get errors with the path because some of the files that are needed are not in the forte repository:

> running build_ext

>  Forte compilation options

    BUILD_TYPE = Debug
    AMBITPATH = home/joana/anaconda3/envs
    MAX_DET_ORB = 64
    ENABLE_CODECOV = OFF
    CMAKE_CONFIG_OPTIONS = 
    CMAKE_BUILD_OPTIONS = 
    NPROCS = 16



> CMake Error: The source directory "/home/joana/forte/command." does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
Traceback (most recent call last):
  File "setup.py", line 156, in <module>
    setup(
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
    return distutils.core.setup(**attrs)
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 177, in setup
    return run_commands(dist)
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 193, in run_commands
    dist.run_commands()
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 968, in run_commands
    self.run_command(cmd)
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/site-packages/setuptools/dist.py", line 1217, in run_command
    super().run_command(command)
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 987, in run_command
    cmd_obj.run()
  File "setup.py", line 58, in run
    self.build_extension(ext)
  File "setup.py", line 116, in build_extension
    subprocess.check_call(["cmake"] + cmake_args)
  File "/home/joana/anaconda3/envs/psi4env/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '"psi4-dev"', 'via', '`conda', 'install', 'psi4-dev', 'psi4[/label/dev]`,', 'then', 'reissue', 'command.', '-Dambit_DIR=home/joana/anaconda3/envs/share/cmake/ambit', '-DCMAKE_BUILD_TYPE=Debug', '-DMAX_DET_ORB=64', '-DENABLE_CODECOV=OFF', '-DENABLE_ForteTests=TRUE']' returned non-zero exit status 1.

**Comment by RenkeHuang** [comment]
Could you try the instructions in this tutorial http://www.evangelistalab.org/2022/03/building-psi4-and-forte-in-2022-on-a-mac-arm64/
I suspect current conda psi4 and ambit are not compatible with forte.

**Comment by jfraxanet** [comment]
Thank you! After a few more attempts I managed to install Forte correctly.
 The problem was that I needed to configure the pythonpath for Psi4. For me, the following worked:
- Download and compile Psi4 using conda https://psicode.org/psi4manual/master/conda.html#faq-psi4deps 
- Configure the paths for PsiAPS https://psicode.org/psi4manual/master/build_planning.html#how-to-configure-paths-for-psiapi
- Download and compile Forte via setup.py
